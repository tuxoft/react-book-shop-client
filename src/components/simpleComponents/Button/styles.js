import styled from "styled-components";

export const Button = styled.button.attrs({ type: "button" })`
  padding: 0.5rem;
  border: 1px solid #f1f1f1;
  background-color: #fff;
  font-size: 1rem;
  border-radius: 5px;
  cursor: pointer;
  margin: auto 0;
`;
