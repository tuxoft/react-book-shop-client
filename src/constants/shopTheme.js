const shopTheme = {
    THEME_COLOR: '#69c3e8',
    DARK_COLOR: '#2598c6'

}