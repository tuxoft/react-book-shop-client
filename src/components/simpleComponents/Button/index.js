import React from "react";
import PropTypes from "prop-types";
import * as styles from "./styles";

const Button = ({ children, ...restProps }) => (
  <styles.Button {...restProps}>{children}</styles.Button>
);

Button.propTypes = {
  onClick: PropTypes.func
};

export default Button;
