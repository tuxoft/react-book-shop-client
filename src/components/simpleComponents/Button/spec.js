import React from "react";
import Button from "./index";
import {ThemeProvider} from "styled-components";

import { shallow, mount } from "enzyme";

import shopTheme from '../../../constants/shopTheme';
it("renders text", () => {
  const button = mount(
      <ThemeProvider theme = {shopTheme}>
        <Button>Button</Button>
      </ThemeProvider>);
  expect(button.text()).toBe("Button");
});

it("renders text", () => {
  const onClickHandler = jest.fn();
  const button = mount(
      <ThemeProvider theme = {shopTheme}>
        <Button onClick={onClickHandler}>Button</Button>
      </ThemeProvider>);
  button.simulate("click");
  expect(onClickHandler).toHaveBeenCalled();
});

it("default button", () => {
  const button = shallow(
      <ThemeProvider theme = {shopTheme}>
        <Button>Button</Button>
      </ThemeProvider>);
  expect(button).toMatchSnapshot();
});

it("primary button", () => {
  const button = shallow(
   <ThemeProvider theme = {shopTheme}>
    <Button primary>Button</Button>
  </ThemeProvider>);
  expect(button).toMatchSnapshot();
});

it("small button", () => {
  const button = shallow(
      <ThemeProvider theme = {shopTheme}>
        <Button small>Button</Button>
      </ThemeProvider>);
  expect(button).toMatchSnapshot();
});
