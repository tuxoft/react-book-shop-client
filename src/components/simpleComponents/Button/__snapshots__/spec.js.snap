// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default button 1`] = `
<Button>
  Button
</Button>
`;

exports[`primary button 1`] = `
<Button
  primary={true}
>
  Button
</Button>
`;

exports[`small button 1`] = `
<Button
  small={true}
>
  Button
</Button>
`;
